sycl/ReleaseNotes.md

```diff
@@ -1,3 +1,296 @@
+- Updated [`sycl_ext_oneapi_complex_algorithms`](sycl/doc/extensions/proposed/sycl_ext_oneapi_complex_algorithms.asciidoc)
+  extension to include `sycl::complex` as supported type for algorithms. [07c5b48f]
+- Clarified sub-group size calculation in [`sycl_ext_oneapi_invoke_simd`](sycl/doc/extensions/experimental/sycl_ext_oneapi_invoke_simd.asciidoc) extension spec. [9b33ad0f]
+- Updated [`sycl_ext_oneapi_accessor_properties`](sycl/doc/extensions/supported/sycl_ext_oneapi_accessor_properties.asciidoc)
+  to mark `has_property` API as `noexcept`. [7805aa3f]
+- Updated [`sycl_ext_intel_device_info`](sycl/doc/extensions/supported/sycl_ext_intel_device_info.md)
+  to support querying free device memory. [0eeef2b3]
+- Updated [`sycl_ext_oneapi_matrix`](sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix.asciidoc)
+  with description of new matrix features. [770f540d]
+- Moved [`sycl_ext_oneapi_invoke_simd`](sycl/doc/extensions/experimental/sycl_ext_oneapi_invoke_simd.asciidoc)
+  extensions specification from `proposed` to `experimental` because
+  implementation is available. [6bee3440]
+
+## Improvements
+
+### SYCL Library
+
+- Ensured that a correct `errc` thrown for an unassociated placeholder
+  accessor. [4f9935ad]
+- Removed dependency on OpenCL ICD Loader from the runtime. [90e8b5ef]
+- Added support for `ZEBIN` format to persistent caching mechanism. [34dcf83d]
+- Added identification mechanism for binaries in newer `ZEBIN` format. [f4dee549]
+- Switched to use `struct` information descriptors in accordance with SYCL 2020.
+  Removed some deprecated information queries. [b3cbda57]
+- Updated `kernel_device_specific::max_sub_group_size` query to match SYCL 2020
+  spec. Deprecated the old variant. [7842d056]
+- Deprecated SYCL 1.2.1 device selectors. [c058380f]
+- Improved error messages reported for unsupported device partitioning. [1c9ddbaa]
+- Made `device` and `platform` default to `default_selector_v`. [b32dd41a]
+- Deprecated `address_space::constant_space`. [351b123a]
+- Marked `sycl::exception::has_context` as `noexcept`. [ad923c9a]
+- Improved range reductions performance on CPU. [3323da65]
+- Made `sycl::exception` `nothrow` copy constructible. [289e33da]
+- Marked `has_property` methods as `noexcept`. [417b5a29]
+- Improved `sycl::event::get_profiling_info` exception message when `event` is
+  default constructed. [2e86cd459]
+- Added a diagnostic (in form of `static_assert`) about kernel lambda size
+  mismatch between host and device. [d278c671] [ec179b7a] [f417a88e]
+- Updated `pipes` class to throw exceptions if used on host. [eab29696]
+- Updated ESIMD Emulator PI plugin to report support for `cl_khr_fp64`
+  extension. [398571a5]
+- Updated Level Zero plugin to prefer copy engine for memory read/write
+  operations. [65c3ea29]
+- Optimized some memory transfers. [92d35cd1]
+- Enabled event caching in Level Zero PI plugin. [a41b33c3]
+- Optimized some reductions for discrete GPUs. [c22a5d3f]
```

Comment:
@aelovikov-intel, I put "some reductions" here, because the commit message only says about some internal helper function. Do you have any suggestions how to improve the wording by stating which exact reductions were optimized?

Reply:
The only thing I can suggest is to change to "some reduction with parallel_for accepting sycl::range" which is a bit more specific but still has some vagueness in it.